apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx

```diff
@@ -112,21 +118,131 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
       isInstantMeeting: eventData && queryIsInstantMeeting ? true : false,
       themeBasis: null,
       orgBannerUrl: team.parent?.bannerUrl ?? "",
-      teamMemberEmail: await getTeamMemberEmail(eventData, email as string),
+      teamMemberEmail: await handleGettingTeamMemberEmail(query, eventTypeId, eventData),
     },
   };
 };
 
-async function getTeamMemberEmail(
-  eventData: {
-    id: number;
-    isInstantEvent: boolean;
-    schedulingType: SchedulingType | null;
-    metadata: Prisma.JsonValue | null;
-    length: number;
-  },
-  email?: string
-): Promise<string | null> {
+interface EventData {
+  id: number;
+  isInstantEvent: boolean;
+  schedulingType: SchedulingType | null;
+  metadata: Prisma.JsonValue | null;
+  length: number;
+}
+
+async function handleGettingTeamMemberEmail(
+  query: ParsedUrlQuery,
+  eventTypeId: number,
+  eventData: EventData
+) {
+  if (
+    !query.email ||
+    typeof query.email !== "string" ||
+    eventData.schedulingType !== SchedulingType.ROUND_ROBIN
+  )
+    return null;
+
+  // Check if a routing form was completed and an routing form option is enabled
+  if (
+    ROUTING_FORM_RESPONSE_ID_QUERY_STRING in query &&
+    Object.values(query).some((value) => value === "true")
+  ) {
+    const { email, skipContactOwner } = await handleRoutingFormOption(query, eventTypeId);
+
+    if (skipContactOwner) return null;
+    if (email) return email;
+  } else {
+    return await getTeamMemberEmail(eventData, query.email);
+  }
+
+  return null;
+}
+
+async function handleRoutingFormOption(query: ParsedUrlQuery, eventTypeId: number) {
+  const nullReturnValue = { email: null, skipContactOwner: false };
+
+  if (typeof query.email !== "string") return nullReturnValue;
+
+  const routingFormQuery = await prisma.app_RoutingForms_Form.findFirst({
```

Comment:
The app specific options live in the routing form data. I didn't want to leak any Salesforce specific information in the URL